src/main/java/org/broadinstitute/hellbender/tools/spark/sv/discovery/BreakpointComplications.java

```diff
@@ -100,56 +145,152 @@ boolean isDupAnnotIsFromOptimization() {
+            initForInsDel(chimericAlignment, contigSeq);
         }
     }
 
-    private void initForSimpleInversion(final AlignmentInterval firstContigRegion, final AlignmentInterval secondContigRegion,
-                                        final byte[] contigSeq) {
-        homologyForwardStrandRep = getHomology(firstContigRegion, secondContigRegion, contigSeq);
-        insertedSequenceForwardStrandRep = getInsertedSequence(firstContigRegion, secondContigRegion, contigSeq);
+    private void initForInversion(final ChimericAlignment chimericAlignment, final byte[] contigSeq) {
+
+        final AlignmentInterval firstAlignmentInterval  = chimericAlignment.regionWithLowerCoordOnContig;
+        final AlignmentInterval secondAlignmentInterval = chimericAlignment.regionWithHigherCoordOnContig;
+
+        homologyForwardStrandRep = getHomology(firstAlignmentInterval, secondAlignmentInterval, contigSeq);
+        insertedSequenceForwardStrandRep = getInsertedSequence(firstAlignmentInterval, secondAlignmentInterval, contigSeq);
         dupSeqRepeatUnitRefSpan = null;
         dupSeqRepeatNumOnRef = dupSeqRepeatNumOnCtg = 0;
+        dupSeqStrandOnRef = dupSeqStrandOnCtg = null;
         cigarStringsForDupSeqOnCtg = null;
         dupAnnotIsFromOptimization = false;
         hasDuplicationAnnotation = false;
     }
 
+    private void initForInvDup(final ChimericAlignment chimericAlignment, final byte[] contigSeq) {
+
+        final AlignmentInterval firstAlignmentInterval  = chimericAlignment.regionWithLowerCoordOnContig;
+        final AlignmentInterval secondAlignmentInterval = chimericAlignment.regionWithHigherCoordOnContig;
+
+        // TODO: 8/8/17 this might be wrong regarding how strand is involved, fix it
+        insertedSequenceForwardStrandRep = getInsertedSequence(firstAlignmentInterval, secondAlignmentInterval, contigSeq);
+        hasDuplicationAnnotation = true;
+
+        dupSeqRepeatNumOnRef = 1;
+        dupSeqRepeatNumOnCtg = 2;
+        dupSeqStrandOnRef = DEFAULT_INV_DUP_REF_STRAND;
+
+        // jump start and jump landing locations
+        final int jumpStartRefLoc = firstAlignmentInterval.forwardStrand ? firstAlignmentInterval.referenceSpan.getEnd()
+                                                                         : firstAlignmentInterval.referenceSpan.getStart();
+        final int jumpLandingRefLoc = secondAlignmentInterval.forwardStrand ? secondAlignmentInterval.referenceSpan.getStart()
+                                                                            : secondAlignmentInterval.referenceSpan.getEnd();
+
+
+
+        if (firstAlignmentInterval.forwardStrand) {
+            final int alpha = firstAlignmentInterval.referenceSpan.getStart(),
+                      omega = secondAlignmentInterval.referenceSpan.getStart();
+            dupSeqRepeatUnitRefSpan = new SimpleInterval(firstAlignmentInterval.referenceSpan.getContig(),
```

Comment:
This is dependent on the two reference span intervals overlapping if I'm reading this right; it might be nice to document or validate that in this method to remind the reader/maintainer.

Reply:
Yes, that is why this method is called `initForInvDup()`, hence assuming the input has "significant" overlaps on their ref spans. Documented.